tortoise/backends/mysql/client.py

```diff
@@ -0,0 +1,205 @@
+
+        self.template = {
+            'host': self.host,
+            'port': self.port,
+            'user': self.user,
+            'password': self.password,
+        }
+
+        self._db_pool = None
+        self._connection = None
+
+        self._transaction_class = type(
+            'TransactionWrapper', (TransactionWrapper, self.__class__), {}
+        )
+
+    async def create_connection(self):
+        try:
+            if not self.single_connection:
+                self._db_pool = await aiomysql.create_pool(db=self.database, **self.template)
+            else:
+                self._connection = await aiomysql.connect(db=self.database, **self.template)
+            self.log.debug(
+                'Created connection with params: '
+                'user={user} database={database} host={host} port={port}'.format(
+                    user=self.user, database=self.database, host=self.host, port=self.port
+                )
+            )
+        except pymysql.err.OperationalError:
+            raise DBConnectionError(
+                "Can't connect to MySQL server: "
+                'user={user} database={database} host={host} port={port}'.format(
+                    user=self.user, database=self.database, host=self.host, port=self.port
+                )
+            )
+
+    async def close(self):
+        if not self.single_connection:
+            self._db_pool.close()
+        else:
+            self._connection.close()
+
+    async def db_create(self):
+        single_connection = self.single_connection
+        self.single_connection = True
+        self._connection = await aiomysql.connect(
+            **self.template
+        )
+        await self.execute_script(
+            'CREATE DATABASE {}'.format(self.database)
+        )
+        self._connection.close()
+        self.single_connection = single_connection
+
+    async def db_delete(self):
+        single_connection = self.single_connection
+        self.single_connection = True
+        self._connection = await aiomysql.connect(
+            **self.template
+        )
+        try:
+            await self.execute_script('DROP DATABASE {}'.format(self.database))
+        except pymysql.err.DatabaseError:
+            pass
+        self._connection.close()
+        self.single_connection = single_connection
+
+    def acquire_connection(self):
+        if not self.single_connection:
+            return self._db_pool.acquire()
+        else:
+            return ConnectionWrapper(self._connection)
+
+    def _in_transaction(self):
+        if self.single_connection:
+            return self._transaction_class(connection=self._connection)
+        else:
+            return self._transaction_class(pool=self._db_pool)
+
+    async def execute_query(self, query):
+        # Temporarily use this method to make mysql work:
+        #  replace VARCHAR to CHAR
+        query = re.sub(r'CAST\((.*?) AS VARCHAR\)', r'CAST(\1 AS CHAR)', query)
```

Comment:
What is the story here? Why it has to be casted in such way? 
Although, if it's inevitable to leave it like this - I think regex should be precompiled

Reply:
I had the same question, then disabled it and ran the tests.
Turns out MySQL can't do case-insensitive compares on a VARCHAR, but only on a CHAR.
Which is downright bizzare.

So I left the workaround in. @etzelwu Do you have more to add?

Reply:
I don't really like that workaround, I'll try to come up with something better to change casting on stage of making query